source/agora/node/Registry.d

```diff
@@ -459,7 +460,7 @@ public class NameRegistry: NameRegistryAPI
     public void onAcceptedBlock (in Block, bool)
         @safe
     {
-        auto validators_zone = this.zones[this.validators];
+        auto validators_zone = this.validators in this.zones;
```

Comment:
This is not safer BTW, because D dereferences the pointer automatically. The safe way would include a null check.

Reply:
My reasoning was that they are initialized in ctor which would make the null check superfluous. It might have been better to assert(0), if they turn out to be null tho.

Reply:
I'm not sure I follow ? `validators_zone` is now a `ZoneData*`. If we know the zone is always present, why the switch from `id[key]` to `key in id` ?

Reply:
To avoid copying it